backend/src/services/integration-auth/integration-sync-secret.ts

```diff
@@ -3751,17 +3750,25 @@ const syncSecretsCloudflarePages = async ({
   );
 
   const metadata = z.record(z.any()).parse(integration.metadata);
-  if (metadata.shouldAutoRedeploy) {
-    await request.post(
-      `${IntegrationUrls.CLOUDFLARE_PAGES_API_URL}/client/v4/accounts/${accessId}/pages/projects/${integration.app}/deployments`,
-      {},
-      {
-        headers: {
-          Authorization: `Bearer ${accessToken}`,
-          Accept: "application/json"
+  if (metadata.shouldAutoRedeploy && integration.targetEnvironment === "production") {
```

Comment:
why does it have to be prod env? can't they choose to sync dev secrets too

Reply:
the way we do auto-deployment right now is via:
`post /accounts/{account_id}/pages/projects/{project_name}/deployment`

which is only for triggering production deployments. if integration sync is supposed to be targetting "preview" environment then it shouldn't trigger a production deployment. 

<img width="657" alt="image" src="https://github.com/user-attachments/assets/ccc0ef71-ce4e-4bc5-9262-f2356de4a1d9" />

Reply:
this also somehow contributed to the error 304 - when you have one integration synced to preview env and one integration synced to production env in CF, then they would both attempt to redeploy even though only the prod integration would be valid